### PR TITLE
Adjust Question API on frontend and backend

### PR DESCRIPTION
**Pull request by WZWren**

### CODE REVIEW

File: peerprep/app/questions/new/page.tsx [hunk 1]
```diff
@@ -132,32 +132,32 @@ function NewQuestion({}: Props) {
           disabled={loading}
           label="Description: "
           name="description"
-          value={formData.description}
+          value={formData.content}
```
Comment (WZWren) [inline peerprep/app/questions/new/page.tsx:135]: Referencing `handleFormTextInput` at Line 66, the name of this `FormTextInput` needs to be renamed to match the struct's field - so name needs to be renamed `content`.

File: peerprep/app/questions/[question]/question.tsx [hunk 2]
```diff
@@ -77,34 +71,16 @@ function QuestionBlock({ question }: Props) {
...
+          {question.topicTags.length == 0 ? (
+            <p>No topics listed.</p>
           ) : (
-            question.categories.map((elem, idx) => <p key={idx}>{elem}</p>)
+            question.topicTags.map((elem, idx) => <p key={idx}>{elem}</p>)
           )}
         </div>
-        <p>{question.description}</p>
+
+        { <div dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(question.content)}} /> }
```
Comment (WZWren) [inline peerprep/app/questions/[question]/question.tsx:82]: Why do we need to DOMPurify this? I don't quite follow why we are storing the HTML tags in content directly.

**Review by WZWren — COMMENTED** [review]
A bit unclear on the changes... what are we storing in `Content`, `Schemas` and `TitleSlug`, and why was the Question struct refactored?

**Comment by rertyy** [comment]
In this db what mongo is storing is 
image

Title is the human readable format
Slug is the machine-readable format
Schemas is for SQL questions if I'm not wrong
Content is the full question, including test cases, in HTML format, hence sanitising
They're all carryovers from the leetcode api

File: peerprep/app/questions/new/page.tsx (hunk 1 above)
Comment (rertyy) [reply peerprep/app/questions/new/page.tsx:135]: ?

File: peerprep/app/questions/[question]/question.tsx (hunk 2 above)
Comment (rertyy) [reply peerprep/app/questions/[question]/question.tsx:82]: Carryover from leetcode api

Comment (WZWren) [reply peerprep/app/questions/[question]/question.tsx:82]: Then we probably need a way to inject the HTML input for making a new question for formatting - fairly certain it should be fine still without modifications though. I'll raise an issue with a nice-to-have for now.

**Pull request merged**